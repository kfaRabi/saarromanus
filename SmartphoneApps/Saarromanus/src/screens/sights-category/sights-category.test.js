import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import SightsCategoryScreen from './index';

const mockStore = configureStore();

const mockFn = jest.fn();
const navigation = {
	navigate: mockFn,
};

const initialState = {
	config: { checkForUpdate: false },
	sightCategories: [
		{
			id: 'all',
			en: {
				name: 'All',
			},
			de: {
				name: 'Alle',
			},
			fr: {
				name: 'Toutes',
			},
		},
		{
			id: '2DEZplgST07PTogH9JWX',
			en: {
				name: 'Siedlungen',
			},
			de: {
				name: 'Settlements',
			},
			fr: {
				name: 'Règlements',
			},
		},
		{
			id: 'JbyYyCCSm9ZT2yKGhjTC',
			en: {
				name: 'Roman Villas',
			},
			de: {
				name: 'Römische Villen',
			},
			fr: {
				name: 'Villas romaines',
			},
		},
		{
			id: 'qXG2ZqlHWLQiDLYea1qn',
			en: {
				name: 'Sanctuaries',
			},
			de: {
				name: 'Heiligtümer',
			},
			fr: {
				name: 'Sanctuaires',
			},
		},
	],
};

let wrapper;
let store;
let component;

beforeAll(() => {
	store = mockStore(initialState);
	wrapper = shallow(
		<SightsCategoryScreen store={store} navigation={navigation} />
	).dive();
	component = wrapper.dive();
});

describe('<SightsCategoryScreen />', () => {
	it('renders properly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('has one main <View /> component', () => {
		expect(component.find('View').length).toBe(1);
	});

	describe('Category List', () => {
		it('should be an array of categories', () => {
			expect(Array.isArray(wrapper.props().sightCategories)).toBe(true);
		});
		it('length should be equal to the number of sights categories received from the store', () => {
			expect(wrapper.props().sightCategories.length).toBe(
				initialState.sightCategories.length
			);
		});
		describe('Single Category', () => {
			it('should contain the component: <SingleCategory />', () => {
				expect(
					component.find('withNavigation(SingleCategory)').length
				).not.toBe(0);
			});
			it('length should equal to the number of categories', () => {
				expect(
					component.find('withNavigation(SingleCategory)').length
				).toBe(initialState.sightCategories.length);
			});
		});
	});
});
