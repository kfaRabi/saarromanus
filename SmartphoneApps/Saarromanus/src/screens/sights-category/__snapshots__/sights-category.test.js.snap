// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SightsCategoryScreen /> renders properly 1`] = `
<SightsCategoryScreen
  checkUpdateStatus={false}
  getSightCategories={[Function]}
  navigation={
    Object {
      "navigate": [MockFunction],
    }
  }
  populateSightCategories={[Function]}
  sightCategories={
    Array [
      Object {
        "de": Object {
          "name": "Alle",
        },
        "en": Object {
          "name": "All",
        },
        "fr": Object {
          "name": "Toutes",
        },
        "id": "all",
      },
      Object {
        "de": Object {
          "name": "Settlements",
        },
        "en": Object {
          "name": "Siedlungen",
        },
        "fr": Object {
          "name": "Règlements",
        },
        "id": "2DEZplgST07PTogH9JWX",
      },
      Object {
        "de": Object {
          "name": "Römische Villen",
        },
        "en": Object {
          "name": "Roman Villas",
        },
        "fr": Object {
          "name": "Villas romaines",
        },
        "id": "JbyYyCCSm9ZT2yKGhjTC",
      },
      Object {
        "de": Object {
          "name": "Heiligtümer",
        },
        "en": Object {
          "name": "Sanctuaries",
        },
        "fr": Object {
          "name": "Sanctuaires",
        },
        "id": "qXG2ZqlHWLQiDLYea1qn",
      },
    ]
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
